parsers/US_SPP.py

```diff
@@ -297,6 +328,109 @@ def fetch_wind_solar_forecasts(
     return data
 
 
+def fetch_live_exchange(
+    zone_key1: str,
+    zone_key2: str,
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+
+    data = get_data(EXCHANGE_URL, session)
+
+    data = data.dropna(axis=0)
+    data["GMTTime"] = pd.to_datetime(data["GMTTime"], utc=True)
+    data = data.loc[data["GMTTime"] <= target_datetime]
+    data = data.set_index("GMTTime")
+
+    exchanges = format_exchange_data(
+        data=data, zone_key1=zone_key1, zone_key2=zone_key2, logger=logger
+    )
+    return exchanges
+
+
+def fetch_historical_exchange(
+    zone_key1: str,
+    zone_key2: str,
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+
+    filename = target_datetime.strftime("TieFlows_%b%Y.csv")
+    file_url = HISTORICAL_EXCHANGE_URL + filename
+
+    data = get_data(file_url, session)
+
+    data["GMTTIME"] = pd.to_datetime(data["GMTTIME"], utc=True)
+    data = data.loc[
+        (data["GMTTIME"] >= target_datetime - timedelta(days=1))
+        & (data["GMTTIME"] <= target_datetime)
+    ]
+    data = data.set_index("GMTTIME")
+
+    exchanges = format_exchange_data(
+        data=data, zone_key1=zone_key1, zone_key2=zone_key2, logger=logger
+    )
+    return exchanges
+
+
+def format_exchange_data(
+    data: pd.DataFrame,
+    zone_key1: str,
+    zone_key2: str,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    """format exchanges data into list of data points"""
+    sorted_zone_keys = "->".join(sorted([zone_key1, zone_key2]))
+    data = data[[col for col in EXCHANGE_MAPPING]]
+    data = data.melt(var_name="zone_key2", value_name="exchange", ignore_index=False)
+    data.zone_key2 = data.zone_key2.map(EXCHANGE_MAPPING)
+
+    data_filtered = data.loc[data["zone_key2"] == zone_key2]
+    data_filtered = data_filtered.groupby([data_filtered.index])["exchange"].sum()
+
+    all_data_points = []
+    for dt in data_filtered.index:
+        data_dt = data_filtered.loc[data_filtered.index == dt]
+        data_point = {
+            "sortedZoneKeys": sorted_zone_keys,
+            "netFlow": round(data_dt.values[0], 4),
+            "datetime": arrow.get(dt).datetime,
+        }
+        all_data_points += [data_point]
```

Comment:
```suggestion
        all_data_points.append(data_point)
```
The way it's written is confusing, I think this would result in a list of list of datapoints. Is it the intention?
Like it would be all_data_points = [[{dp1}], [{dp2}]] while I think we rather want [{dp1}, {dp2}].